src/rtp/rtcp/twcc.rs

```diff
@@ -515,13 +515,19 @@ fn build_interims(
             }
         }
 
-        let diff_time = if r.time < prev.1 {
+        let diff_time: Result<i32, _> = if r.time < prev.1 {
             // negative
             let dur = prev.1 - r.time;
-            -(dur.as_micros() as i32)
+            dur.as_micros().try_into().map(|m: i32| -m)
         } else {
             let dur = r.time - prev.1;
-            dur.as_micros() as i32
+            dur.as_micros().try_into()
+        };
+
+        let Ok(diff_time) = diff_time else {
+            // This time diff was so large we can't even reliably represent it. Treat the same
+            // as the more sane out-of-range case below.
+            break;
```

Comment:
`Duration::as_micros()` gives us as 128 bit number. I think it is cleaner to cast to an i64 and rely on the rule below rather than introducing this `Result`.

Another idea is to keep the i128, however I have a hunch it's faster to cast to something that has native support on the CPU. In fact, that's probably why we are here to start off with.

Reply:
It's not clear to me that an i64 would solve the issue - my impression is that it would just push the problematic range further out. It would be great if we could just `as i128` and be done with it, but since `as_micros()` specifically returns `u128` there's a danger zone around `u128::MAX`.

Of course, as we go from `i32` to `i64` to `i128`, the problematic cases get more and more unrealistic 😅 But despite the cleanliness hit, if nothing else I think there's value in the explicitness, signaling to readers that we've thought about this case.

Reply:
As far as performance concerns of an explicitly checked cast, I'm confident that this is an easy case for a branch predictor, making it basically free.

Reply:
> the problematic cases get more and more unrealistic 😅

This is the point, yes. Microseconds represented as 32 bit will handle a duration of ~4300 seconds or ~70 minutes. Having such a delta in this context (twcc) would be madness, but let's assume it can happen if you sleep the laptop or something. For 64 bit we get a duration of ~580 thousand years, at which point the battery of my laptop will be drained.

> As far as performance concerns of an explicitly checked cast

I wasn't thinking about the performance of your solve, but rather about casting to 64 bit and then perform those divisions. Dividing 128 bit on hardware without 128-bit support will be more expensive, hence I propose to cast to 64.

Reply:
Okay then, I'll change the cast to 64.